lib/index.js

```diff
@@ -31,21 +31,21 @@ function Setup(T, { checkTypes , _env=[] }){
     }
 
     function mapConstrToFn(constr) {
-    return (
-        constr === String
-            ? T.String
-        : constr === Number
-            ? T.Number
-        : constr === Boolean
-            ? T.Boolean
-        : constr === Object
-            ? T.Object
-        : constr === Array
-            ? T.Array(T.Any)
-        : constr === Function
-            ? T.AnyFunction
-        : constr
-    )
+        return (
+            constr === String
+                ? T.String
+            : constr === Number
+                ? T.Number
+            : constr === Boolean
+                ? T.Boolean
+            : constr === Object
+                ? T.Object
+            : constr === Array
+                ? T.Array(T.Any)
+            : constr === Function
+                ? T.AnyFunction

```

Comment:
If you want to accept sanctuary-def types, type representatives (such as `Array` and `Function`), and predicates, I suggest providing three separate functions for constructing a sum type. As a user I find it confusing that `$.String` and `String` can be used more or less interchangeably (as can `Array` and `Array.isArray`). That said, it is (unfortunately in my view) idiomatic for JavaScript functions to accept values of all sorts of different types. ;)